src/systems/filecoin_mining/storage_mining/storage_miner_actor.go

```diff
@@ -27,41 +43,50 @@ func (cs *ChallengeStatus_I) OnChallengeResponse(currEpoch block.ChainEpoch) Cha
 	return cs
 }
 
-func (cs *ChallengeStatus_I) isChallenged() bool {
+func (cs *ChallengeStatus_I) IsChallenged() bool {
 	// true (isChallenged) when LastChallengeEpoch is later than LastChallengeEndEpoch
 	return cs.LastChallengeEpoch() > cs.LastChallengeEndEpoch()
 }
 
+func (st *StorageMinerActorState_I) _isChallenged(rt Runtime) bool {
+	return st.ChallengeStatus().IsChallenged()
+}
+
+func (a *StorageMinerActorCode_I) _isChallenged(rt Runtime) bool {
+	h, st := a.State(rt)
+	ret := st._isChallenged(rt)
+	h.Release(st)
```

Comment:
I think that `h.Release` and `h.Commit` should mangle `st` and `h` such that all further invocations on them panic. something like this:

```
func (h *ActorStateHandle) Release(st util.Any) {
  // whatever
  h.destroy() // all further invocations to h functions panic
  st.destroy() // all further invocations to st functions panic
}
```
It may be easy to achieve by making `h` and `st` be set to nil (if they're holding on to pointers, could work -- all function invocations would try to dereference nil and panic). But not sure

Reply:
```go
// can passing in address let `Release` set `st = nil` and achieve the intent?
h.Release(&st)


func (h *ActorStateHandle) Release(st &util.Any) {
  // whatever
  h.Runtime = nil // all further invocations to h functions panic
  st = nil // all further invocations to h functions panic
}
```

Reply:
That would cause the functions to panic with a nil dereference, which we might not want -- if this is done by a runtime check, they can instead exit via `rt.Fatal`, causing the transaction to abort but not necessarily causing the interpreter to crash. Not totally clear; which would be preferable?